src/transforms/prune_items.rs

```diff
@@ -41,48 +42,100 @@ pub fn prune_items<F: FnMut(&syn::File) -> bool>(file: &mut syn::File, mut try_c
 }
 
 trait Prune {
-	fn prune(&mut self, level: usize, index: &mut usize) -> bool;
+    fn prune(&mut self, level: usize, index: &mut usize) -> bool;
 }
 
 impl Prune for Vec<syn::Item> {
-	fn prune(&mut self, level: usize, index: &mut usize) -> bool {
-		if level == 0 {
-			if *index < self.len() {
-				self.remove(*index);
-				true
-			} else {
-				*index -= self.len();
-				false
-			}
-		} else {
-			for item in self {
-				if match item {
-					syn::Item::Mod(syn::ItemMod { content: Some((_, items)), .. })
-						=> items.prune(level - 1, index),
-					syn::Item::Impl(syn::ItemImpl { items, .. })
-						=> items.prune(level - 1, index),
-					_ => false
-				} {
-					return true
-				}
-			}
-			false
-		}
-	}
+    fn prune(&mut self, level: usize, index: &mut usize) -> bool {
+        if level == 0 {
+            if *index < self.len() {
+                self.remove(*index);
+                true
+            } else {
+                *index -= self.len();
+                false
+            }
+        } else {
+            for item in self {
+                if match item {
+                    syn::Item::Mod(syn::ItemMod {
+                        content: Some((_, items)),
+                        ..
+                    }) => items.prune(level - 1, index),
+                    syn::Item::Struct(item @ syn::ItemStruct{ .. }) => item.prune(level - 1, index),
+                    syn::Item::Impl(syn::ItemImpl { items, .. }) => items.prune(level - 1, index),
+                    syn::Item::Enum(item @ syn::ItemEnum{ .. }) => item.prune(level - 1, index),
+                    _ => false,
+                } {
+                    return true;
+                }
+            }
+            false
+        }
+    }
 }
 
 impl Prune for Vec<syn::ImplItem> {
-	fn prune(&mut self, level: usize, index: &mut usize) -> bool {
-		if level == 0 {
-			if *index < self.len() {
-				self.remove(*index);
-				true
-			} else {
-				*index -= self.len();
-				false
-			}
-		} else {
-			false
-		}
-	}
+    fn prune(&mut self, level: usize, index: &mut usize) -> bool {
+        if level < 5 {
+            if *index < self.len() {
+                self.remove(*index);
+                true
+            } else {
+                *index -= self.len();
+                false
+            }
+        } else {
+            false
+        }
+    }
 }
+
+impl Prune for syn::ItemEnum{
+    fn prune(&mut self, level: usize, index: &mut usize) -> bool {
+        if level < 5 {
+            if *index < self.variants.len() {
+                //  dbg!(my_enum.variants.clone());
+                let mut smaller : Punctuated<syn::Variant, syn::Token![,]> = Punctuated::new();
+                for (i, pair) in self.variants.pairs().enumerate() {
```

Comment:
You can avoid the `clone` by using `variants.into_iter().enumerate()`. But you need to swap it out of `self` first. So the whole thing becomes:
```rust
let orig = std::mem::swap(&mut self.variants, Punctuated::new());
for (i, variant) in orig.into_iter().enumerate() {
    if i != *index {
        self.variants.push(variant);
    }
}
```